### PR TITLE
Add sorting functions

### PR DESCRIPTION
**Pull request by Drakulix**
~~Incomplete. This is missing documentation and tests.~~

@iwburns What do you think about this implementation?

### CODE REVIEW

**Comment by Drakulix** [comment]
Changes discussed in https://github.com/iwburns/id-tree/issues/3#issuecomment-267186883 are added via ca5aac3

**Comment by iwburns** [comment]
Alright, I like this solution very much!  Good call on using `mem::swap`!

This is just a nitpick (and it's partially targeted at myself), but how do you feel about the name `sort_children_by_data`?  I just named it that earlier because I was in a hurry and it sounded alright, but now I'm thinking `sort_children` would be better since it follows the pattern [here](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.sort) a bit better with `sort`, `sort_by_key`, and `sort_by`.

What do you think?

**Comment by Drakulix** [comment]
I actually had(!) this function named `sort_children` at first.
But I felt, that this would not make absolutely clear what key we are sorting.
`NodeId`s / Indices? Oldest `Node` first? `data()`?

I think `sort_children_by_data` is a little bit more declarative, if you encounter it in the code, but I have no strong feelings about this. `sort_children` is also perfectly fine, just not my personal preference.

**Comment by iwburns** [comment]
Yeah, those are fair points.
Let's keep it as `sort_children_by_data` then.  I think that's clearer as well.

**Comment by Drakulix** [comment]
Added some documentation with tests.
Is this sufficient, or do you want failure tests for the sorting functions as well?

**Comment by iwburns** [comment]
If you don't mind adding the failure tests as well I would appreciate it.
Otherwise, this looks great to me!

**Comment by Drakulix** [comment]
Sure thing

**Comment by Drakulix** [comment]
Rebased and tests added.

File: src/tree.rs [hunk 1]
```diff
@@ -436,6 +438,147 @@ impl<T> Tree<T> {
...
+    /// tree.sort_children_by(&root_id, |a, b| a.data().cmp(b.data()));
+    ///
+    /// # for (i, id) in tree.get(&root_id).unwrap().children().iter().enumerate() {
+    /// #   assert_eq!(*tree.get(&id).unwrap().data(), i as i32);
+    /// # }
+    /// ```
+    ///
+    pub fn sort_children_by<F>(&mut self, node_id: &NodeId, mut compare: F) -> Result<(), NodeIdError>
+        where F: FnMut(&Node<T>, &Node<T>) -> Ordering
+    {
```
Comment (Drakulix) [inline src/tree.rs]: Any comments on the style here @iwburns ?
I kinda dislike same line bracket when using `where`, but its totally up to you. Should I change it?

Comment (iwburns) [inline src/tree.rs]: Nope, I agree with you, it looks quite strange with a `where` clause.  You have it exactly how I would have done it.

Also, could you point me to the docs for rustfmt?  I followed your link earlier (and did some searching as well), but couldn't find anything other than their README.md.  I'm specifically looking for what can be configured in the rustfml.toml file.

Comment (Drakulix) [inline src/tree.rs]: https://github.com/rust-lang-nursery/rustfmt#configuring-rustfmt

They don't keep a compiled list around, you have to run the command with the argument listed.

Comment (iwburns) [inline src/tree.rs]: Oh, haha, that's what I get for reading too fast....sorry about that, and thanks!

**Comment by iwburns** [comment]
@Drakulix v0.2.7 is up on crates.io

**Pull request merged**